snuba/clickhouse/translators/snuba/allowed.py

```diff
@@ -0,0 +1,87 @@
+from typing import TypeVar, Union
+
+from snuba.clickhouse.translators.snuba import SnubaClickhouseStrictTranslator
+from snuba.datasets.plans.translator.mapper import ExpressionMapper
+from snuba.query.expressions import (
+    Argument,
+    Column,
+    CurriedFunctionCall,
+    FunctionCall,
+    Lambda,
+    Literal,
+    SubscriptableReference,
+)
+
+TExpIn = TypeVar("TExpIn")
+TExpOut = TypeVar("TExpOut")
+
+
+class SnubaClickhouseMapper(
+    ExpressionMapper[TExpIn, TExpOut, SnubaClickhouseStrictTranslator]
+):
+    pass
+
+
+# Below are the abstract classes that need to be extended in order to write a mapping rule
```

Comment:
I have been quite arbitrarily restrictive on this file. A lot more than we actually need. Likely we will release some of these constraint when we start writing real translation rules. But I suspect it is not going to change much.